core/templates/components/state-editor/state-responses-editor/state-responses.component.ts

```diff
@@ -151,16 +151,15 @@ export class StateResponsesComponent implements OnInit, OnDestroy {
   }
 
   isSelfLoopThatIsMarkedCorrect(outcome: Outcome): boolean {
-    if (!outcome ||
-        !this.stateEditorService.getCorrectnessFeedbackEnabled()) {
+    if (outcome && typeof outcome === 'object') {
```

Comment:
Why do you need the `typeof` check? It wasn't there before.

Reply:
I agree typeof is not needed as on the frontend the input is also type of Outcome and I added this as this was being followed in other functions also

Reply:
Remove it, and yes undefined is a valid input because in state-responses.component, we are not initializing the value of defaultOutcome, and it is initialized in the ngOnInit function, which takes some time.

Reply:
That seems problematic. Why is this function being called before ngOnInit is called? I thought ngOnInit is supposed to be the first thing called when a component is initialized.

Reply:
Yes, ngOnInit is called after the component is initialized (the execution flow is constructor -> ngOnChanges -> ngOnInit). 

We subscribe to various events and set outcome when an event is emitted. I guess that can be a possible reason.